apps/opik-backend/src/main/java/com/comet/opik/domain/stats/StatsMapper.java

```diff
@@ -28,6 +28,9 @@ public static ProjectStats mapProjectStats(Row row, String entityCountLabel) {
                 .add(new ProjectStats.CountValueStat("metadata", row.get("metadata", Long.class)))
                 .add(new ProjectStats.AvgValueStat("tags", row.get("tags", Double.class)));
 
+        var totalEstimatedCost = row.get("total_estimated_cost", Double.class);
```

Comment:
I recall discussing with @andrescrz regarding this, and decision was to use BigDecimal for all related to cost/money. I guess for consistency it would be good to follow it here as well.

Reply:
The issue is that the other AVGs use double because Clickhouse uses returns Float64 for such cases. So, in this case, BigDecimal will no affect precision as the value is already calculated using Float64, and we will have to change the contract type.

Reply:
This is the field definition in ClickHouse: `total_estimated_cost Decimal64(8)`.

So there's just an issue with `avgIf` function. Just check the ClickHouse documentation or quickly investigate to get a `Decimal` results from the ClickHouse avg operation and you should be good to go.

Alternatives:
- Use another flavour of the avg function, which returns a Decimal instead of Float64.
- Use ClickHouse converstion function from float to decimal , specially if they control rounding.
- Use ClickHouse rounding function returning a decimal, and control de rounding.

As mentioned, by default we use use decimal types for money calculation, with banking rounding by default.

Reply:
Done